syntax = "proto3";

message Input {
	string document = 1;
}

message IndexPair {
	int32 index_start = 2;
	int32 index_end = 3;
}

message TopSpans {
	repeated IndexPair top_span_pair = 4;
}

message PredictedAntecedents {
	repeated RepeatedInt predicted = 5;
}

message RepeatedInt{
	int32 value = 6;
}

message TokenizedDocument {
	repeated Document document_token = 6;
}

message Document{
	string value = 8;
}

message Coref {
	repeated IndexPair corefs_pair = 7;
}

message Cluster {
	repeated Coref corefs = 8;
}

message Output{
	TopSpans top_spans= 1;
	PredictedAntecedents predicted_antecedents= 2;
	TokenizedDocument document= 3;
	Cluster clusters = 4;
}

service CorefResolution {
	rpc resolve (Input) returns (Output){}
}